examples/load_and_reindex_bboxes.py

```diff
@@ -0,0 +1,336 @@
+        s=5,
+        color=list_colors[0],
+    )
+
+    # plot future centroid positionsin white
+    ax.scatter(
+        x=ds.position[data_idx + 1 : data_end_idx, 0, 0].data,
+        y=ds.position[data_idx + 1 : data_end_idx, 0, 1].data,
+        s=5,
+        color="white",
+    )
+
+    ax.set_title(f"Frame {ds.time[data_idx].item()}")
+    ax.set_xlabel("x (pixles)")
+    ax.set_ylabel("y (pixels)")
+    ax.set_xlabel("")
+
+fig.tight_layout()
+
+
+# %%
+# Fill in empty values with forward filling
+# ----------------------------------------------------
+# We can fill in the frames with missing values for the  ``position`` and
+# ``shape`` arrays by taking the last valid value in time. In this way, a
+# box's position and shape stay constant if for a current frame the box
+# has no annotation defined.
+
+ds_ff = ds.reindex(
+    {"time": list(range(ds.time[-1].item()))},
+    method="ffill",  # propagate last valid index value forward
+)
+
+# check the first 14 frames of the data
+print("Position data array (first 14 frames):")
+print(ds_ff.position.data[:14, 0, :])  # time, individual, space
+
+print("----")
+print("Shape data array (first 14 frames):")
+print(ds_ff.shape.data[:14, 0, :])  # time, individual, space
+
+# %%
+# Fill in empty values with NaN
+# ----------------------------------------------------
+# Alternatively, we can fill in the missing frames with NaN values.
+# This can be useful if we want to interpolate the missing values later.
+ds_nan = ds.reindex(
+    {"time": list(range(ds.time[-1].item()))},
+    method=None,  # default
+)
+
+# check the first 14 frames of the data
+print("Position data array (first 14 frames):")
+print(ds_nan.position.data[:14, 0, :])
+
+print("----")
+print("Shape data array (first 14 frames):")
+print(ds_nan.shape.data[:14, 0, :])
+
+# %%
+# Linearly interpolate NaN values
+# ----------------------------------------------------------
+# We can instead fill in the missing values in the dataset applying linear
+# interpolation to the ``position`` and ``shape`` data arrays. In this way,
+# we would be assuming that the centroid of the bounding box moves linearly
+# between the two annotated values, and its width and height change linearly
+# as well.
+#
+# We use the dataset with NaN values as an input to the
+# ``interpolate_over_time`` function.
+ds_interp = ds_nan.copy()
+
+for data_array_str in ["position", "shape"]:
+    ds_interp[data_array_str] = interpolate_over_time(
+        data=ds_interp[data_array_str],
+        method="linear",
+        max_gap=None,
+        print_report=False,
+    )
+
+# check the first 14 frames of the data
+print("Position data array (first 14 frames):")
+print(ds_interp.position.data[:14, 0, :])
+
+print("----")
+print("Shape data array (first 14 frames):")
+print(ds_interp.shape.data[:14, 0, :])
+
+
+# %%
+# Compare interpolation methods
+# ------------------------------
+# We can now qualitatively compare the three different methods of filling
```

Comment:
I was confused by this wording. There are only two methods of filling here, ffill, or linear. 
But I then understood you count filling with NaNs as the third method (essentially mark the empty frames as missing).

Maybe you can clarify the wording a bit to convey what the three variants are.

Reply:
agree, I found the wording for this example to be quite awkward myself too 😅